### ISSUE TITLE
False positive in import chains

### ISSUE THREAD

**Issue by arquolo** [python/mypy#8114]
The following valid code yields:
`foo/__init__.py:3:11: error: Name 'bar' is not defined`.
In `foo/bar.py`:
```python
__all__ = ('baz', )

def baz() -> None:
    pass
```
In `foo/__init__.py`:
```python
from .bar import *

__all__ = bar.__all__
```
However code is fully valid, so no errors should be raised.
Using mypy `0.750` from PyPI and Python `3.7.5`.
Run via `mypy --strict foo`

**Comment by ethanhs** [comment]
It is possible mypy didn't process things in the correct order, what command line did you run?

**Comment by arquolo** [comment]
I run it via `mypy --strict foo.py` in the directory where `foo` is.

**Comment by JelleZijlstra** [comment]
The subtlety here is that `bar` is not actually set directly in `__init__.py`, but at runtime it magically does exist. I'm guessing mypy just doesn't reflect the runtime behavior here.

**Comment by arquolo** [comment]
Also found that this also triggers the false positive:
`foo/bar.py` is the same.
`foo/__init__.py`:
```python
from foo.bar import *
__all__ = bar.__all__
```
or
```python
from foo.bar import baz
__all__ = bar.__all__
```
or
```python
from .bar import baz
__all__ = bar.__all__
```

**Comment by msullivan** [comment]
> However code is fully valid, so no errors should be raised.
It is not the philosophy or goal of mypy to accept all fully valid Python code, though. We're happy to be opinionated (balanced against pragmatism, of course).

I would like to err on the side of supporting *less* of Python's subtle name and import semantics, not more, I think.

If you also add `from . import bar`, you should be able to get this to work.

Thanks for the report!

**Comment by arquolo** [comment]
So to suppress false positive, I need to add another import from already imported module?
